docs/usage.rst

```diff
@@ -108,14 +108,14 @@ looking for modules that begin with a specific prefix.
 You can look for either or both, in single call to
 :meth:`~PluginManager.discover`, which will import any modules or entry_points
 that follow one of the aforementioned conventions, and search them for
-functions decorated with the appropriate :class:`HookImplementationMarker` (as shown
-above in ``some_plugin.py``)
+functions decorated with the appropriate :class:`HookImplementationMarker` (as
+shown above in ``some_plugin.py``)
 
 .. code-block:: python
 
-   plugin_manager.discover(
-      entry_point='my_project.plugin', prefix='my_project_'
-   )
+   plugin_manager.discover_entry_point = 'my_project.plugin'
+   plugin_manager.discover_prefix = 'my_project_'
+   plugin_manager.discover()
```

Comment:
Why is this an improvement?

Reply:
Also, as far as I can tell, the actual code hasn't changed, so, for me, the previous, more functional form was significantly nicer...

Reply:
Ah, I take it back, it has changed, I just had a filter on the commits I was seeing. @tlambert03 has this been released? And in general can we start adopting code review for this project?

Reply:
Sure absolutely, sorry about that. No this isn’t released yet.  If you want to add back the option to accept values to discover() we can do that.  The reasoning was that a) it makes it much easier to call as a callback/slot and b) the places we are looking for plugins rarely change for a given plugin manager, so using the class attributes made sense.  

The previous behavior used the class attributes as an override and I can go back to that if you want. It would also need to be propagated to the new iter_available method

Reply:
I agree that it makes the docs a bit uglier

Reply:
@Jni, let me know if you want me to open up a new PR with some reversions.
I enabled mandatory reviews on this repo

Reply:
When you say “used class attributes as an override”, do you mean the opposite? Used them as default but allowing passing others? I think a happy middle is to allow passing in the parameters but making them optional. I think a PR for that would get merged pretty quickly. =P

Reply:
yes, indeed, I mean the exact opposite :). that's why I shouldn't respond at 5am.  I can definitely add that back.

here's one use-case I want to point out, which is I think what swayed me towards making it stateful.  This PR added the [`PluginManager.iter_available()`](https://github.com/napari/napari-plugin-engine/pull/3/files#diff-e9e4cb84ed8d8b365fcf368cb1a1d998R161-R173) iterator, which is in turn used by `discover()`.  That was useful for https://github.com/napari/napari/pull/1357, where we could see what "would be" registered, before actually do so.  Adding back the optional parameters of course doesn't change that, but it does require that someone call both `iter_available()` and `discover()` with the same parameters, and allows for the possibility that they could make a mistake.  That's obviously then a user error, but I think that's what led me to remove the parameters.  (minor issue, just wanted to mention)